dockerfile/cuda11.1.1.dockerfile

```diff
@@ -85,6 +89,19 @@ RUN cd /tmp && \
     cd /tmp && \
     rm -rf nccl
 
+# Install Docker and NVIDIA Container Toolkit
+RUN curl -fsSL https://download.docker.com/linux/ubuntu/gpg | \
+    gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg && \
+    echo \
```

Comment:
could you follow https://github.com/docker-library/docker/blob/f30c3dd6f0665b2b34e520f9e9348dcaa8755de9/20.10/Dockerfile#L32?

and move it after apt-get block because it won't be modified frequently
```dockerfile
ENV DOCKER_VERSION=20.10.8
RUN cd /tmp && \
    wget https://download.docker.com/linux/static/stable/x86_64/docker-${DOCKER_VERSION}.tgz -O docker.tgz && \
    tar --extract --file docker.tgz --strip-components 1 --directory /usr/local/bin/ && \
    rm docker.tgz
```

Reply:
we don't need docker-ce, containerd, etc. here, only several binaries to parse `docker` command and communicate with host daemon

Reply:
Thanks, it's much cheaper, before this, I followed the docker installation tutorials from https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository